### PR TITLE
[#51229] Remove feature flags for storage_primer_design and storage_one_drive_integration

### PR DESCRIPTION
**Pull request by ba1ash**
https://community.openproject.org/work_packages/51229

### CODE REVIEW

**Review by apfohl — APPROVED** [review]
Looks very clean now with all the components.

File: modules/storages/spec/features/admin_storages_spec.rb [hunk 1]
```diff
@@ -537,7 +537,8 @@
     end
   end
 
-  it 'creates, edits and deletes storages', :webmock do
+  # skipped to be revised later. broken due to removal of storage_primer_design feature flag
+  xit 'creates, edits and deletes storages', :webmock do
```
Comment (akabiru) [inline modules/storages/spec/features/admin_storages_spec.rb:541]: @ba1ash we can delete these specs, they're for the old flow

File: modules/storages/spec/features/admin_storages_spec.rb [hunk 2]
```diff
@@ -753,7 +754,8 @@
     end
     let!(:unconfigured_storage) { create(:nextcloud_storage) }
 
-    it 'reports storages that are not configured correctly' do
+    # skipped to be revised later. broken due to removal of storage_primer_design feature flag
+    xit 'reports storages that are not configured correctly' do
```
Comment (akabiru) [inline modules/storages/spec/features/admin_storages_spec.rb:758]: Ditto: would also need deleting the configuration checks component

File: modules/storages/spec/features/admin_storages_spec.rb (hunk 1 above)
Comment (ba1ash) [reply modules/storages/spec/features/admin_storages_spec.rb:541]: Sure, I will do it. I postponed it because I was not sure. Thanks for clarifying!

**Pull request merged**